README.md

```diff
@@ -1,289 +1,610 @@
-
-**Note** - a few updates / enhancements are being proposed to the current `ServiceBinding` CR, tracked [here](https://github.com/application-stacks/service-binding-specification/issues/16#issuecomment-605309629).
-
-Sample CR
+## Example Directory Structure
+
+```plain
+$SERVICE_BINDING_ROOT
+├── account-database
+│   ├── type
+│   ├── connection-count
+│   ├── uri
+│   ├── username
+│   └── password
+└── transaction-event-stream
+    ├── type
+    ├── connection-count
+    ├── uri
+    ├── certificates
+    └── private-key
 ```
+
+# Service Binding
+
+A Service Binding describes the connection between a [Provisioned Service](#provisioned-service) and an [Application Projection](#application-projection).  It is codified as a concrete resource type.  Multiple Service Bindings can refer to the same service.  Multiple Service Bindings can refer to the same application.
+
+Restricting service binding to resources within the same namespace is strongly **RECOMMENDED**.  Cross-namespace service binding **SHOULD** be secured appropriately by the implementor to prevent attacks like privilege escalation and secret enumeration.
+
+A Service Binding resource **MUST** define a `.spec.application` which is an `ObjectReference` to a `PodSpec`-able resource.  A Service Binding resource **MUST** define a `.spec.service` which is an `ObjectReference` to a Provisioned Service-able resource.  A Service Binding resource **MAY** define a `.spec.name` which is the name of the service when projected into the application.
+
+A Service Binding resource **MUST** define a `.status.conditions` which is an array of `Condition` objects.  A `Condition` object **MUST** define `type`, `status`, and `lastTransitionTime` entries.  At least one condition containing a `type` of `Ready` must be defined.  The `status` of the `Ready` condition **MUST** have a value of `True`, `False`, or `Unknown`.  The `lastTranstionTime` **MUST** contain the last time that the condition transitioned from one status to another.  A Service Binding resource **MAY** define `reason` and `message` entries to describe the last `status` transition.
+
+## Resource Type Schema
+
+```yaml
 apiVersion: service.binding/v1alpha1
 kind: ServiceBinding
 metadata:
-  name: example-service-binding
+  name:                 # string
 spec:
-  mountPathPrefix: "/var/bindings"
-  services:
-    - group: postgres.dev
-      kind: Service
-      name: global-user-db
-      version: v1beta1
-      id: postgres-global-user
-    - group: ibmcloud.ibm.com
-      version: v1alpha1
-      kind: Binding
-      name: coligo-service-binding
-      id: coligo-service-binding
-  application:
-    name: nodejs-rest-http-crud
-    group: apps
-    version: v1
-    resource: deployments
+  name:                 # string, optional, default: .metadata.name
+  type:                 # string, optional
```

Comment:
I'm a little concerned with the `type` be able to be specified in the `ServiceBinding` CR. Let's say i have a library that iterates over `$SERVICE_BINDINGS_ROOT` directory and looks for binding info of a known service(ex: kafka) and makes them accessible inside the application, and lets say if the `type` is specified as `xyz` in CR, then this library can never make the binding info accessible to the application. Should the `type` be provided by backing service operator?

I'm thinking of a library like this https://github.com/pivotal-cf/java-cfenv/blob/master/java-cfenv-boot/src/main/java/io/pivotal/cfenv/spring/boot/AmqpCfEnvProcessor.java

Reply:
The `type` field here is an option to set/override the type already defined in the binding Secret. It's primarily to help compatibility with existing service providers that don't define this information. The ideal is absolutely for the service provider to define the type within the referenced Secret, in that case the field in the ServiceBinding CR would be empty.

Reply:
@yharish991 - the recommendation is definitely that `type` is provided by the backing service operator.  The spec says:  `The Secret SHOULD contain a type entry with a value that identifies the abstract classification of the binding`

So while it's recommending that it should contain, it could be empty, which is one of the reasons it's beneficial to be able to set that in the `ServiceBinding` CR (enables frameworks as you have described).  

I guess another use case would be if you have applications that are expecting the `type` to be of a certain format or contain certain text, then you can override what the backing service has provided.

Reply:
oops, i missed that, there have been too many changes, trying to catch up..